tests/test-books-api.js

```diff
@@ -0,0 +1,178 @@
+require('dotenv').config()
+const dbbooks = require('../dbbooks')
+const assert = require('assert')
+const axios = require('axios')
+const dbusers = require('../dbusers')
+
+const State = {
+    SUCCESS: "Success",
+    ERROR: "Error"
+}
+const ErrorResponseMessage = "Error occurs when execute query on database"
+
+const isbn = '123456sdfa123'
+const name = 'Bleach'
+const author = 'Tite Kubo'
+const publisher = 'Kim Dong'
+const quantity = 145
+const id_category = '13fbb02e-9d06-48de-9ff4-276ff5959e1f'
+
+describe('books api', async() => {
+    let book_id = null
+    before(async() => {
+        const result = await dbbooks.create_book_async(isbn, name, author, publisher, quantity, id_category)
+        book_id = result.rows[0].new_id
+    })
+
+    describe('get books', async() => {
+        it('1. should return status 200 and the number of data response is equal to the number of books when called correctly', async() => {
+            let response = await axios.get(`${process.env.API_URL}/books`)
+            let apiBooksCount = response.data.length
+            const result = await dbbooks.count_number_of_books_async()
+            const dbBooksCount = result.rows[0].count
+            assert.strictEqual(apiBooksCount, parseInt(dbBooksCount))
+            assert.strictEqual(response.status, 200)
+        })
+    })
+
+    describe('get book by ID', async() => {
+        it('2. should return status 200 and the data response correct user when called', async() => {
+            let response = await axios.get(`${process.env.API_URL}/books/${book_id}`)
+            let apiBook = response.data[0]
+            assert.strictEqual(apiBook.id, book_id)
+            assert.strictEqual(apiBook.name, name)
+            assert.strictEqual(apiBook.isbn, isbn)
+            assert.strictEqual(apiBook.author, author)
+            assert.strictEqual(apiBook.publisher, publisher)
+            assert.strictEqual(apiBook.quantity, quantity)
+            assert.strictEqual(apiBook.id_category, id_category)
+
+            assert.strictEqual(response.status, 200)
+        })
+
+        it('3. should return status 500 and the data response will give the error message when called with the wrong id format', async() => {
+            let response = await axios.get(`${process.env.API_URL}/books/1`).catch(error => {
+                assert.strictEqual(error.response.data.State, State.ERROR)
+                assert.strictEqual(error.response.data.ErrorMessage, ErrorResponseMessage)
+                assert.strictEqual(error.response.status, 500)
+            })
+        })
+
+        it('4. should return status 200 and the data response will give a null array when called with the wrong id\'s value', async() => {
+            let new_id = null;
+            if(book_id.charAt(35) == '1') {
```

Comment:
There is an easier way to change Id, but can we 100% sure the changed Id hasn't already in the database?

Reply:
I fixed it